kernel/environ.ml

```diff
@@ -1012,68 +1003,58 @@ struct
   let find_opt env key m = M.find_opt (Q.canonize env key) m
 end
 
-module QConstant =
-struct
-  type t = Constant.t
-  let equal _env c1 c2 = Constant.CanOrd.equal c1 c2
-  let compare _env c1 c2 = Constant.CanOrd.compare c1 c2
-  let hash _env c = Constant.CanOrd.hash c
-  let canonize _env c = Constant.canonize c
-end
+module type QNameS =
```

Comment:
I don't think it's a good idea to merge the name API with the map API. Can't we keep both separate via some inclusion trickery?